### PR TITLE
WIP: First version of the CI/CD scenario

### PR DESCRIPTION
**Pull request by christophetd**
As discussed with @RyanJarv.

To review/discuss:
* For the TODOs: I'm afraid to spoil the solution if I replace them with the exploitation route. What do you think? As someone who used CloudGoat, it's sometimes disappointed to see the solution when trying to spin up the scenario.
* Can you try running the end-to-end tests to make sure they're easy to run for someone else as well?

### CODE REVIEW

File: scenarios/cicd/README.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
+# Scenario: cicd
+
+**Size:** Medium
```
Comment (christophetd) [inline scenarios/cicd/README.md:3]: Size and difficulty TBD. I don't have enough knowledge of the other scenarios to judge

File: scenarios/cicd/start.sh [hunk 2]
```diff
@@ -0,0 +1 @@
+#!/bin/bash
```
Comment (RyanJarv) [inline scenarios/cicd/start.sh:1]: I think this can b removed if not needed.

**Review by RyanJarv — CHANGES_REQUESTED** [review]
Thanks for the PR! Still reviewing this but will post the comments I have so far.

File: scenarios/cicd/terraform/sdlc.tf [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  ]
+
+  provisioner "local-exec" {
+    working_dir = path.module
+    on_failure  = fail
+    interpreter = ["bash", "-c"]
+    environment = {
+      AWS_REGION = var.region,
+      AWS_DEFAULT_REGION = var.region
+    }
```
[the remark this answers is not in the document]
Comment (christophetd) [reply scenarios/cicd/terraform/sdlc.tf:27]: It works fine as-is either manually or through cloudgoat

File: scenarios/cicd/assets/src/.gitignore [hunk 4]
```diff
@@ -0,0 +1 @@
+buildspec.old.yml
```
[the remark this answers is not in the document]
Comment (christophetd) [reply scenarios/cicd/assets/src/.gitignore:1]: This is needed, it's the file with credentials I push to the git history.

File: scenarios/cicd/start.sh (hunk 2 above)
Comment (christophetd) [reply scenarios/cicd/start.sh:1]: Yeah it's just a good practice I generally follow to ensure my scripts get interpreted by bash (and not another shell like sh)

File: scenarios/cicd/terraform/cheatsheet.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
+
+
+## Step 1
+
+You are given credentials of an initial IAM user. 
+
```
[the remark this answers is not in the document]
Comment (christophetd) [reply scenarios/cicd/terraform/cheatsheet.md:6]: good catch, will solve in next commit

File: scenarios/cicd/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (christophetd) [reply scenarios/cicd/README.md:3]: good catch, will solve in next commit

File: scenarios/cicd/cheat_sheet.md [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+- Find the CodeCommit user ID of `cloner` using `aws iam list-ssh-public-keys --user-name cloner --output text --query 'SSHPublicKeys[0].SSHPublicKeyId'`
+
+- Use the following SSH configuration (in your `.ssh/config`):
+
+```
+Host *.amazonaws.com
+	IdentityFile ~/.ssh/stolen_key
+```
+
+- Then, clone the repository using `git clone ssh://APKA2...@git-codecommit.eu-west-1.amazonaws.com/v1/repos/backend-api`
```
[the remark this answers is not in the document]
Comment (christophetd) [reply scenarios/cicd/cheat_sheet.md]: solved in 
57804cb

File: scenarios/cicd/terraform/README.md [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+
+FooCorp is a company exposing a public-facing API. Customers of FooCorp submit sensitive data to the API every minute to the following API endpoint:
+
+```
+POST {apiUrl}/prod/hello
+Host: {apiHost}
+Content-Type: text/html
+
+superSecretData=...
+```
```
[the remark this answers is not in the document]
Comment (christophetd) [reply scenarios/cicd/terraform/README.md]: solved in 65c5fe4

File: scenarios/cicd/start.sh (hunk 2 above)
Comment (RyanJarv) [reply scenarios/cicd/start.sh:1]: Oh I guess I meant the start.sh file I don't think needs to exist if it's empty, but I could be wrong about that.

**Review by RyanJarv — APPROVED** [review]

**Comment by RyanJarv** [comment]
Really appreciate the effort that went into this. Especially like the simulated requests to the API and it's great to have some testing now.

I'll follow up on this soon to get CI running those tests on merge to master as well.

**Comment by RyanJarv** [comment]
FYI I updated the main README.md with the description from the manifest, if you want to change that just let me know.

https://github.com/RhinoSecurityLabs/cloudgoat/blob/master/README.md#cicd-medium--moderate

**Pull request merged**